src/instructlab/config/init.py

```diff
@@ -0,0 +1,124 @@
+import click
+from instructlab import helpers
+from instructlab import utils
+from git import Repo, GitError
+from os.path import exists, dirname, basename, splitext
+from os import listdir
+
+@click.command()
+@click.option(
+    "--interactive/--non-interactive",
+    default=True,
+    show_default=True,
+    help="Initialize the environment assuming defaults.",
+)
+@click.option(
+    "--model-path",
+    type=click.Path(),
+    default=helpers.DEFAULT_MODEL_PATH,
+    show_default=True,
+    help="Path to the model used during generation.",
+)
+@click.option(
+    "--taxonomy-base",
+    default=helpers.DEFAULT_TAXONOMY_BASE,
+    show_default=True,
+    help="Base git-ref to use when listing/generating new taxonomy.",
+)
+@click.option(
+    "--taxonomy-path",
+    type=click.Path(),
+    default=helpers.DEFAULT_TAXONOMY_PATH,
+    show_default=True,
+    help=f"Path to {helpers.DEFAULT_TAXONOMY_REPO} clone.",
+)
+@click.option(
+    "--repository",
+    default=helpers.DEFAULT_TAXONOMY_REPO,
+    show_default=True,
+    help="Taxonomy repository location.",
+)
+@click.option(
+    "--min-taxonomy",
+    is_flag=True,
+    help="Shallow clone the taxonomy repository with minimum size. "
+    "Please do not use this option if you are planning to contribute back "
+    "using the same taxonomy repository. ",
+)
+def init(
+    interactive,
+    model_path,
+    taxonomy_path,
+    taxonomy_base,
+    repository,
+    min_taxonomy,
+):
+    """Initializes environment for InstructLab"""
+    clone_taxonomy_repo = True
+    if interactive:
+        if exists(helpers.DEFAULT_CONFIG):
+            overwrite = click.confirm(
+                f"Found {helpers.DEFAULT_CONFIG} in the current directory, do you still want to continue?"
+            )
+            if not overwrite:
+                return
+        click.echo(
+            "Welcome to InstructLab CLI. This guide will help you to setup your environment."
+        )
+        click.echo(
+            "Please provide the following values to initiate the "
+            "environment [press Enter for defaults]:"
+        )
+
+        taxonomy_path = utils.expand_path(
+            click.prompt("Path to taxonomy repo", default=taxonomy_path)
+        )
+
+    try:
+        taxonomy_contents = listdir(taxonomy_path)
```

Comment:
I would prefer this line was left as-is -- `os.listdir()` instead of `listdir()`

Reply:
mostly to minimize unnecessary code change

... secondly a matter of taste. I'm not a big fan of importing everything into the base namespace. I like seeing where something is coming from without having to go back to the top to figure it out.